webv2/utilities/utilities.go

```diff
@@ -221,3 +141,195 @@ func RemoveIndex(Pks []ddl.IndexKey, index int) []ddl.IndexKey {
 
 	return list
 }
+
+func IsTypeChanged(newType, table, colName string, Conv *internal.Conv) (bool, error) {
+
+	//sessionState := session.GetSessionState()
+
+	srcTableName := Conv.ToSource[table].Name
+
+	sp, ty, err := GetType(Conv, newType, table, colName, srcTableName)
+	if err != nil {
+		return false, err
+	}
+	colDef := sp.ColDefs[colName]
+	return !reflect.DeepEqual(colDef.T, ty), nil
+}
+
+func IsPartOfPK(col, table string) bool {
+	sessionState := session.GetSessionState()
+
+	for _, pk := range sessionState.Conv.SpSchema[table].Pks {
+		if pk.Col == col {
+			return true
+		}
+	}
+	return false
+}
+
+func IsPartOfSecondaryIndex(col, table string) (bool, string) {
+	sessionState := session.GetSessionState()
+
+	for _, index := range sessionState.Conv.SpSchema[table].Indexes {
+		for _, key := range index.Keys {
+			if key.Col == col {
+				return true, index.Name
+			}
+		}
+	}
+	return false, ""
+}
+
+func IsPartOfFK(col, table string) bool {
+	sessionState := session.GetSessionState()
+
+	for _, fk := range sessionState.Conv.SpSchema[table].Fks {
+		for _, column := range fk.Columns {
+			if column == col {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// TODO: create a map to store referenced column to get
+// this information in O(1).
+// TODO:(searce) can have foreign key constraints between columns of the same table, as well as between same column on a given table.
```

Comment:
Please check if these comments are still relevant

Reply:
Not relevant as of now. It was part of webv1